nodejs/app.js

```diff
@@ -3,11 +3,19 @@
 const path = require('path');
 const ejs = require('ejs')
 const express = require('express');
-const middleware = require('./middleware/auth');
 
 // Set up the express app.
 const app = express();
 
+// Allow the express app to be exported into other files. 
+module.exports = app;
+
+// Build the conf object from the conf files.
+app.conf = require('./conf/conf');
+
+// Hold onto the auth middleware 
+const middleware = require('./middleware/auth');
```

Comment:
isnt it good practice to keep imports at the top in a group?

Reply:
The `/middleware/auth.js` file depends on `app.conf` existing and being populated and will not run correctly if its imported first.